### ISSUE TITLE
mongolian not able to connect to remote mongo when mongo running on different port. Instead connects to localhost

### ISSUE THREAD

**Issue by sankalpsaxena81** [marcello3d/node-mongolian#110]
mongolian not able to connect to remote mongo when mongo running on different port. It rather connects to localhost . This happens due to code in server.js line no 23,62,63,115-117.
This is quite severe since mongolian does not throw any error but does operation on local mongo instance.

This only happens when remote mongo runs on non default port.

**Comment by marcello3d** [comment]
What does your connection code like?

**Comment by sankalpsaxena81** [comment]
We are using mongo-sync which is a wrapper on top of mongolian.
Code looks like: 

var Server = require("mongo-sync").Server;
server = new Server(remote_host:port);
server.db(db_name).getCollection(collectionName);

If we do console.log for the line numbers 23,62,63,115-117 of server.js we clearly see that it has localhost added to the list of servers.

**Comment by sankalpsaxena81** [comment]
I realized that now it is also happening for remote machine running on default port.So it happens for remote physical machine as well as remote virtual machines. This is scary since the code does not throw any exception but silently adds localhost server to its list.

**Comment by marcello3d** [comment]
This looks odd. Looking at the code in server.js, localhost is only used if no server is specified. Does your URL not match the regexp `/^(?:mongo(?:db)?:\/\/)?(?:([^:]+):([^@]+)@)?(.+?)(?::([0-9]+))?(?:\/(.*))?$/`?

I'm assuming your code looks like `new Server("foohost.com:20000")`?

**Comment by sankalpsaxena81** [comment]
Sorry!! This is not an issue .

We figured out why mongolian was using localhost when given remote mongo. Actually remote box was configured to be mongo replica set. All instances of mongo in replica set running on the same machine. When we did "rs.config()"  it gave all nodes with hostname "localhost" and hence mongolian was picking this localhost(I saw in the code that it tries to find primary from the replica set). As soon as we changed localhost to ip address, mongolian started working fine. 

Ideally replica set nodes should be on different machines but since this was just a test env all nodes were running on localhost.
